flink/v2.1/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/ZkLockFactory.java

```diff
@@ -222,4 +252,28 @@ public void unlock() {
       }
     }
   }
+
+  @VisibleForTesting
+  RetryPolicy createRetryPolicy(ZKRetryPolicies policy) {
```

Comment:
This parametrization is strange.
Either:
- Use a static method and add all of the required inputs
- Use a non-static method, then don't add parameters which are already available in an attribute